app/gui2/src/components/ComponentBrowser.vue

```diff
@@ -107,26 +114,23 @@ const originScenePos = computed(() => {
 
 function panIntoView() {
   const origin = originScenePos.value
-  const inputArea = new Rect(
+  const screenRect = cbRoot.value?.getBoundingClientRect()
+  if (!screenRect) return
+  const area = new Rect(
     origin,
-    new Vec2(WIDTH, INPUT_AREA_HEIGHT).scale(clientToSceneFactor.value),
+    new Vec2(screenRect.width, screenRect.height).scale(clientToSceneFactor.value),
   )
-  const panelsAreaDimensions = new Vec2(WIDTH, PANELS_HEIGHT).scale(clientToSceneFactor.value)
-  const panelsArea = new Rect(origin.sub(new Vec2(0, panelsAreaDimensions.y)), panelsAreaDimensions)
-  const vizHeight = VISUALIZATION_HEIGHT * clientToSceneFactor.value
   const margins = scaleValues(PAN_MARGINS, clientToSceneFactor.value)
   props.navigator.panTo([
-    // Always include the bottom-left of the input area.
-    { x: inputArea.left, y: inputArea.bottom },
-    // Try to reach the top-right corner of the panels.
-    { x: inputArea.right, y: panelsArea.top },
-    // Extend down to include the visualization.
-    { y: inputArea.bottom + vizHeight },
+    // Always include the top-left of the input area.
+    { x: area.left, y: area.top },
+    // Try to reach the bottom-right corner of the panels.
+    { x: area.right, y: area.bottom },
     // Top (and left) margins are more important than bottom (and right) margins because the screen has controls across
     // the top and on the left.
-    { x: inputArea.left - margins.left, y: panelsArea.top - margins.top },
+    { x: area.left - margins.left, y: area.top - margins.top },
     // If the screen is very spacious, even the bottom right gets some breathing room.
-    { x: inputArea.right + margins.right, y: inputArea.bottom + vizHeight + margins.bottom },
+    { x: area.right + margins.right, y: area.bottom + margins.bottom },
```

Comment:
While we're modifying this: When this logic was written, the big plus button was in the bottom left; it's in the bottom right now.

Reply:
There is, indeed, a higher margin on the left, but this is still useful to make space for component name (and circular menu),.